### PR TITLE
Support "latest" Terraform version

### PR DESCRIPTION
**Pull request by alexjurkiewicz**
And version specifiers like `1` and `1.2` (picking the most recent matching stable version).

### CODE REVIEW

File: src/executors/default.yml [hunk 1]
```diff
@@ -8,4 +8,4 @@ parameters:
   tag:
     description: Specify the Terraform Docker image tag for the executor
     type: string
-    default: "1.0.0" # update commands/install when updating this
+    default: latest
```
Comment (KyleTryon) [inline src/executors/default.yml]: Could we leave this as 1.0.0 for the time being, that we can include this change as a feature change in the next update coming soon? We can update this to latest in the next major version change we will issue shortly after.

File: src/commands/install.yml [hunk 2]
```diff
@@ -4,8 +4,12 @@ description: |
...
   terraform_version:
     type: "string"
-    description: "Specify version of terraform to install."
-    default: "1.0.0" # should match executor
+    description: |
+      Version of Terraform to install. Only stable versions are supported. You can specify the version in three ways:
+      * latest: most recent version
+      * full version number (1.2.3): a specific version
+      * anything else (1.2, 1): the most recent stable version which matches this prefix.
+    default: latest
```
Comment (KyleTryon) [inline src/commands/install.yml]: Could we leave this as 1.0.0 for the time being, that we can include this change as a feature change in the next update coming soon? We can update this to latest in the next major version change we will issue shortly after.

**Review by KyleTryon — CHANGES_REQUESTED** [review]
Thank you! I appreciate all of the hard work in refactoring. I am going to put the tests though now to see if we can catch anything. Other than that, I am asking for 2 small changes, the same change, that should keep this change a feature, rather than major change and then I can release it very soon. We can then update the defaults shortly after in a major update.

Thank you!

**Comment by KyleTryon** [comment]
For some interesting reason it looks as though the tests failed.. somehow related to RVM. It may help to separate the refactoring from the change to minimize the amount of change.
I am currently working to add in some other PRs and resolve some issues as well, so I apologize for the conflicts that are possible to come.
If i do not hear back from you soon, I will happily take a crack at updating it taking inspiration from your PR.

I think we can deploy the next minor version very soon, this may be the last change we add in before the update.

File: src/executors/default.yml (hunk 1 above)
Comment (alexjurkiewicz) [reply src/executors/default.yml]: done!

File: src/commands/install.yml (hunk 2 above)
Comment (alexjurkiewicz) [reply src/commands/install.yml]: done!

**Comment by alexjurkiewicz** [comment]
Thanks Kyle. I'm stumped too 😕 I've added some debug logging which should trigger if you run the `test-deploy` workflow again 🙏 
Feel free to take over this PR if you prefer.

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/terraform?version=dev:f6a59b101ff6abeabe8c20494acd74c5bbb84a8b

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/terraform?version=dev:1d18dc9ecc3c10d45d0152581117ca1a4d51e3b0

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/terraform?version=dev:49fbceeb92281bb1d0cbddb4d3766c46c6ef63fb

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/terraform?version=dev:b7ae064494589ddcc873ea23569394c91ba0cc79

**Comment by KyleTryon** [comment]
I think we got it :D

**Review by KyleTryon — APPROVED** [review]

**Comment by KyleTryon** [comment]
Many thanks @alexjurkiewicz 🙇

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/terraform?version=dev:911134d278d200e43afc8df6654457f222d185b4

**Comment by alexjurkiewicz** [comment]
Thank you!

**Pull request merged**